### PR TITLE
Fix nonzero exit for conda-build entrypoint

### LINKED ISSUE #5167: conda build has non-zero exit status despite successful build
### What happened?

When building a recipe with conda-build version 24.1.0, the return status is non-zero despite the build being successful. This does not happen with conda-build version 3.28.4.

### Conda Details

<details>
<summary><code>conda info</code></summary>

```shell
     active environment : /…/envs/miniconda3
    active env location : /…/envs/miniconda3
            shell level : 2
       user config file : /…/.condarc
 populated config files : /…/.condarc
          conda version : 23.11.0
    conda-build version : 24.1.0
         python version : 3.11.5.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=zen
                          __conda=23.11.0=0
                          __cuda=12.3=0
                          __glibc=2.38=0
                          __linux=6.6.13=0
                          __unix=0=0
       base environment : /…/miniconda3  (writable)
      conda av data dir : /…/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/sebp/linux-64
                          https://conda.anaconda.org/sebp/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /…/miniconda3/pkgs
                          /…/.conda/pkgs
       envs directories : /…/miniconda3/envs
                          /…/.conda/envs
               platform : linux-64
             user-agent : conda/23.11.0 requests/2.31.0 CPython/3.11.5 Linux/6.6.13-200.fc39.x86_64 fedora/39 glibc/2.38 solver/libmamba conda-libmamba-solver/23.12.0 libmambapy/1.5.3 aau/0.4.3 c/K7pj3PuF7vhwR8S115QX0A s/oxhLIYnOkyClhlSbRCzyKA e/oS9vSrZdGoblDAkkam1dig
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

</details>

<details>
<summary><code>conda config</code></summary>

```shell
add_pip_as_python_dependency: False
auto_update_conda: False
changeps1: False
channel_priority: flexible
channels:
  - sebp
  - defaults
show_channel_urls: True
always_yes: True
report_errors: False
anaconda_upload: False
conda_build:
  pkg_format: 2
```

</details>

<details>
<summary><code>conda list</code></summary>

```shell
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
anaconda-anon-usage       0.4.3           py311hfc0e8ea_100    defaults
anaconda-client           1.12.2          py311h06a4308_0    defaults
archspec                  0.2.1              pyhd3eb1b0_0    defaults
attrs                     23.1.0          py311h06a4308_0    defaults
beautifulsoup4            4.12.2          py311h06a4308_0    defaults
boltons                   23.0.0          py311h06a4308_0    defaults
brotli-python             1.0.9           py311h6a678d5_7    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
c-ares                    1.19.1               h5eee18b_0    defaults
ca-certificates           2023.12.12           h06a4308_0    defaults
certifi                   2023.11.17      py311h06a4308_0    defaults
cffi                      1.16.0          py311h5eee18b_0    defaults
chardet                   4.0.0           py311h06a4308_1003    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
click                     8.1.7           py311h06a4308_0    defaults
conda                     23.11.0         py311h06a4308_0    defaults
conda-build               24.1.0          py311h06a4308_0    defaults
conda-content-trust       0.2.0           py311h06a4308_0    defaults
conda-index               0.4.0              pyhd3eb1b0_0    defaults
conda-libmamba-solver     23.12.0            pyhd3eb1b0_1    defaults
conda-package-handling    2.2.0           py311h06a4308_0    defaults
conda-package-streaming   0.9.0           py311h06a4308_0    defaults
cryptography              41.0.7          py311hdda0065_0    defaults
defusedxml                0.7.1              pyhd3eb1b0_0    defaults
distro                    1.8.0           py311h06a4308_0    defaults
filelock                  3.13.1          py311h06a4308_0    defaults
fmt                       9.1.0                hdb19cb5_0    defaults
icu                       73.1                 h6a678d5_0    defaults
idna                      3.4             py311h06a4308_0    defaults
jinja2                    3.1.3           py311h06a4308_0    defaults
jsonpatch                 1.32               pyhd3eb1b0_0    defaults
jsonpointer               2.1                pyhd3eb1b0_0    defaults
jsonschema                4.19.2          py311h06a4308_0    defaults
jsonschema-specifications 2023.7.1        py311h06a4308_0    defaults
jupyter_core              5.5.0           py311h06a4308_0    defaults
krb5                      1.20.1               h143b758_1    defaults
ld_impl_linux-64          2.38                 h1181459_1    defaults
libarchive                3.6.2                h6ac8c49_2    defaults
libcurl                   8.4.0                h251f7ec_1    defaults
libedit                   3.1.20230828         h5eee18b_0    defaults
libev                     4.33                 h7f8727e_1    defaults
libffi                    3.4.4                h6a678d5_0    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgomp                   11.2.0               h1234567_1    defaults
liblief                   0.12.3               h6a678d5_0    defaults
libmamba                  1.5.3                haf1ee3a_0    defaults
libmambapy                1.5.3           py311h2dafd23_0    defaults
libnghttp2                1.57.0               h2d74bed_0    defaults
libsolv                   0.7.24               he621ea3_0    defaults
libssh2                   1.10.0               hdbd6064_2    defaults
libstdcxx-ng              11.2.0               h1234567_1    defaults
libuuid                   1.41.5               h5eee18b_0    defaults
libxml2                   2.10.4               hf1b16e4_1    defaults
lz4-c                     1.9.4                h6a678d5_0    defaults
markupsafe                2.1.3           py311h5eee18b_0    defaults
menuinst                  2.0.1           py311h06a4308_1    defaults
more-itertools            10.1.0          py311h06a4308_0    defaults
nbformat                  5.9.2           py311h06a4308_0    defaults
ncurses                   6.4                  h6a678d5_0    defaults
openssl                   3.0.13               h7f8727e_0    defaults
packaging                 23.1            py311h06a4308_0    defaults
patch                     2.7.6             h7b6447c_1001    defaults
patchelf                  0.17.2               h6a678d5_0    defaults
pcre2                     10.42                hebb0a14_0    defaults
pip                       23.3.1          py311h06a4308_0    defaults
pkginfo                   1.9.6           py311h06a4308_0    defaults
platformdirs              3.10.0          py311h06a4308_0    defaults
pluggy                    1.0.0           py311h06a4308_1    defaults
psutil                    5.9.0           py311h5eee18b_0    defaults
py-lief                   0.12.3          py311h6a678d5_0    defaults
pybind11-abi              4                    hd3eb1b0_1    defaults
pycosat                   0.6.6           py311h5eee18b_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyopenssl                 23.2.0          py311h06a4308_0    defaults
pysocks                   1.7.1           py311h06a4308_0    defaults
python                    3.11.5               h955ad1f_0    defaults
python-dateutil           2.8.2              pyhd3eb1b0_0    defaults
python-fastjsonschema     2.16.2          py311h06a4308_0    defaults
python-libarchive-c       2.9                pyhd3eb1b0_1    defaults
pytz                      2023.3.post1    py311h06a4308_0    defaults
pyyaml                    6.0.1           py311h5eee18b_0    defaults
readline                  8.2                  h5eee18b_0    defaults
referencing               0.30.2          py311h06a4308_0    defaults
reproc                    14.2.4               h295c915_1    defaults
reproc-cpp                14.2.4               h295c915_1    defaults
requests                  2.31.0          py311h06a4308_0    defaults
requests-toolbelt         1.0.0           py311h06a4308_0    defaults
rpds-py                   0.10.6          py311hb02cf49_0    defaults
ruamel.yaml               0.17.21         py311h5eee18b_0    defaults
setuptools                68.2.2          py311h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
soupsieve                 2.5             py311h06a4308_0    defaults
sqlite                    3.41.2               h5eee18b_0    defaults
tk                        8.6.12               h1ccaba5_0    defaults
tqdm                      4.65.0          py311h92b7b1e_0    defaults
traitlets                 5.7.1           py311h06a4308_0    defaults
truststore                0.8.0           py311h06a4308_0    defaults
tzdata                    2023c                h04d1e81_0    defaults
urllib3                   1.26.18         py311h06a4308_0    defaults
wheel                     0.41.2          py311h06a4308_0    defaults
xz                        5.4.5                h5eee18b_0    defaults
yaml                      0.2.5                h7b6447c_0    defaults
yaml-cpp                  0.8.0                h6a678d5_0    defaults
zlib                      1.2.13               h5eee18b_0    defaults
zstandard                 0.19.0          py311h5eee18b_0    defaults
zstd                      1.5.5                hc292b87_0    defaults
```

</details>

### Additional Context

Example recipe:

```yaml
{% set name = "tqdm" %}
{% set version = "4.66.1" %}
{% set sha256 = "d88e651f9db8d8551a62556d3cff9e3034274ca5d66e93197cf2490e2dcb69c7" %}

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  fn: {{ name }}-{{ version }}.tar.gz
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
  sha256: {{ sha256 }}

build:
  number: 0
  script: {{ PYTHON }} -m pip install -vvv .

requirements:
  host:
    - python
    - pip
    - setuptools_scm
  run:
    - python

about:
  home: https://github.com/tqdm/tqdm
  license: MPL-2.0 AND MIT
  license_file: LICENCE
  summary: 'A Fast, Extensible Progress Bar for Python and CLI'
```

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Resolves #5167 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Comment by mbargull** [comment]
Oh, I didn't even notice the `.execute` functions returned non-`int`s before; whoops!

Could we just let `.cli.main_build.execute` return `0` in any (non-failure) case?
Apart from the tests, I don't think we have consumers of that function that use the return value.

Yes, one could argue about API/deprecation etc..
I don't think this is necessarily needed here -- we do have the `.api` module and if anyone else uses `.cli.main_*.execute` they are knowingly doing bad stuff.
The proposed `returns_outputs` argument would just added unneeded potential maintenance burden, IMO.

**Comment by kenodegard** [comment]
> Could we just let `.cli.main_build.execute` return `0` in any (non-failure) case?
Apart from the tests, I don't think we have consumers of that function that use the return value.

I'd prefer that, was just trying to avoid a larger refactor since this is a patch.

> The proposed `returns_outputs` argument would just added unneeded potential maintenance burden, IMO.

Absolutely, not a fan at all.

@mbargull good feedback, I'll work some more on this

File: conda_build/cli/main_build.py [hunk 1]
```diff
@@ -535,21 +535,20 @@ def execute(args: Sequence[str] | None = None):
...
 
-    outputs = None
     if parsed.output:
         config.verbose = False
         config.quiet = True
         config.debug = False
-        outputs = [output_action(recipe, config) for recipe in parsed.recipe]
+        for recipe in parsed.recipe:
+            output_action(recipe, config)
     elif parsed.test:
```
Comment (mbargull) [inline conda_build/cli/main_build.py:551]: We could also insert an `return 0` here, turn it into an `if parsed.test` and remove the `not parsed.output` in the last `if` at the bottom of the function.

**Review by mbargull — APPROVED** [review]
LGTM!

**Comment by mbargull** [comment]
There are some `sys.exit(x)` instead of `return x` calls in those `execute` functions -- we could clean that up at some point; but not necessary to do in this PR.

File: conda_build/cli/main_build.py (hunk 1 above)
Comment (kenodegard) [reply conda_build/cli/main_build.py]: See https://github.com/conda/conda-build/pull/5169/commits/329bdafb422509452a0363da2422709482a08b9d

**Review by mbargull — APPROVED** [review]

**Pull request merged**